interface-definitions/service_dns_forwarding.xml.in

```diff
@@ -793,6 +793,155 @@
               </node>
+              <tagNode name="zone-to-cache">
+                <properties>
+                  <help>Load a zone into the recursor cache</help>
+                  <valueHelp>
+                    <format>txt</format>
+                    <description>The name of the zone to load</description>
+                  </valueHelp>
+                </properties>
+                <children>
+                  <node name="source">
+                    <properties>
+                      <help>Source to load zone</help>
+                    </properties>
+                    <children>
+                      <leafNode name="axfr">
+                        <properties>
+                          <help>Source IP address</help>
+                          <valueHelp>
+                            <format>ipv4</format>
+                            <description>IPv4 address</description>
+                          </valueHelp>
+                          <constraint>
+                            <validator name="ipv4-address"/>
+                          </constraint>
+                        </properties>
+                      </leafNode>
+                      <leafNode name="url">
+                        <properties>
+                          <help>Source URL</help>
+                          <valueHelp>
+                            <format>url</format>
+                            <description>Set URL location</description>
+                          </valueHelp>
+                          <constraint>
+                            <validator name="url" argument="--scheme http --scheme https"/>
+                          </constraint>
+                        </properties>
+                      </leafNode>
+                    </children>
+                  </node>
+                  <node name="options">
+                    <properties>
+                      <help>Zone to cache options</help>
+                    </properties>
+                    <children>
+                      <leafNode name="timeout">
+                        <properties>
+                          <help>The maximum time a retrieval using the AXFR or URL source may take</help>
+                          <valueHelp>
+                            <format>u32:1-3600</format>
+                            <description>Request timeout in seconds</description>
+                          </valueHelp>
+                          <constraint>
+                            <validator name="numeric" argument="--range 1-3600"/>
+                          </constraint>
+                        </properties>
+                        <defaultValue>20</defaultValue>
+                      </leafNode>
+                      <leafNode name="refresh">
+                        <properties>
+                          <help>The interval to wait between retrievals. A value of zero means the retrieval is done once at startup and on Lua configuration reload</help>
```

Comment:
"Periodic zone retrieval interval"? I'd also name the option `refresh-interval`.

I also find their use of zero rather counter-intuitive. Most often a "zero time interval" in configs means "don't wait". Here its meaning is opposite of the common practice.

Maybe we should only allow non-zero values here and make zero the implicit default. Or make separate options like `refresh on-reload` and `refresh interval <n>`. I'm generally against mutually exclusive options when it's possible, but sometimes it may be the best way to make configs easy to understand.

Also, we could reuse the human-readable time value function from `system task-scheduler` etc., to make it possible to write `3h` or `1d`.